sktime/forecasting/multiplexer.py

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env python3 -u
+# -*- coding: utf-8 -*-
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+
+from sktime.forecasting.base._base import DEFAULT_ALPHA
+from sktime.forecasting.base._sktime import _OptionalForecastingHorizonMixin
+from sktime.forecasting.base._sktime import _SktimeForecaster
+import copy
+
+__author__ = ["Kutay Koralturk"]
+__all__ = ["Multiplexer"]
+
+
+class Multiplexer(_OptionalForecastingHorizonMixin, _SktimeForecaster):
+    def __init__(
+        self,
+        components: dict,
+        select=None,
+    ):
+
+        """
+        Multiplexer facilitates a framework for performing
+        model selection process over different model classes.
+        It should be used in conjunction with ForecastingGridSearchCV
+        to get full utilization.
+
+        Single use of Multiplexer with components and select
+        parameter specified, works just like the selected component.
+        It does not provide any further use in that case.
+
+        When used with ForecastingGridSearchCV, multiplexer
+        provides an ability to compare different model class
+        performances with each other, just like a model tournament.
+        When ForecastingGridSearchCV is fitted with a multiplexer,
+        returned value for the select argument of best_params_
+        attribute of ForecastingGridSearchCV, gives the best
+        performing model class among given models provided in components.
+
+        Parameters
+        ----------
+        components : dict
+            A dictionary composed of key-value pairs
+            of forecaster names and forecaster objects.
+        select: str
+            An argument to make a selection among components.
+            Multiplexer uses select to choose which component to fit.
+            Important for using with ForecastingGridSearchCV as a
+            hyperparameter.
+
+        Attributes
+        ----------
+        _forecaster : Sktime forecaster
+            forecaster that multiplexer will currently
+            forecast with.
+        _forecaster_fit_params: float
+            Fit params for the forecaster that is
+            going to be used by the forecaster in the
+            _forecaster
+        """
+
+        self.components = components
+        self.select = select
+        self._check_components()
+        self._forecaster_fit_params = None
```

Comment:
Yes, I agree with @fkiraly and @aiwalter, best to keep `fit_params` in `fit`

Reply:
are we still passing this in the init?

Reply:
I wouldn't store this in the object